### PR TITLE
feat: Render Custom widget for standalone array fields

### PR DESCRIPTION
**Pull request by alfonsoar**
### Reasons for making this change

This PR should fix https://github.com/rjsf-team/react-jsonschema-form/issues/1385

It allows developers to specify a array in the schema without having to specify an `enum` or `uniqueItems`. 

### Checklist

* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests
  - [x] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [x] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/master/CHANGELOG.md) with a description of the PR
* [x] **I'm updating documentation**
  - [x] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding a new feature**
  - [x] I've updated the playground with an example use of the feature

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -40,6 +40,12 @@ should change the heading of the (upcoming) version to include a major version b
 - Differentiated the material-ui 4 and 5 themes
 - Added chakra-ui theme
 
+
+# v3.4.0 (Upcoming)
```
Comment (alfonsoar) [inline CHANGELOG.md:44]: Is this the next release?

**Comment by alfonsoar** [comment]
There are a ton of `package-lock.json` changes. Not sure if that is expected but it's a bit risky. Should we just omit them?

**Review by AbigailFernandes — APPROVED** [review]

**Review by epicfaace — COMMENTED** [review]
Documentation we need to change:
- https://react-jsonschema-form.readthedocs.io/en/latest/advanced-customization/custom-templates/#arrayfieldtemplate - add a note at beginning that mentions you can use ui:widget for arrays as well
- https://react-jsonschema-form.readthedocs.io/en/latest/advanced-customization/custom-widgets-fields/#adding-your-own-custom-widgets - add "array" as a type that supports widgets
- https://react-jsonschema-form.readthedocs.io/en/latest/usage/arrays/#multiple-choice-list - put "MCL" under a new section called "custom widgets", also talk about how you can use a custom widget in general for an array

File: CHANGELOG.md [hunk 2]
```diff
@@ -40,8 +40,10 @@ should change the heading of the (upcoming) version to include a major version b
 - Differentiated the material-ui 4 and 5 themes
 - Added chakra-ui theme
 
+# v3.4.0
 ## @rjsf/core
 - Introduce `idSeparator` prop to change the path separator used to generate field names (https://github.com/rjsf-team/react-jsonschema-form/pull/2628)
+- Array fields support custom widgets without `enums` or `uniqueItems`
```
Comment (epicfaace) [inline CHANGELOG.md:46]:
```suggestion
- Array fields support custom widgets (previously, only multiple-choice arrays with `enums` or `uniqueItems` support it)
```

**Review by epicfaace — COMMENTED** [review]
Revert package-lock.json changes, keep them locally if needed

File: packages/core/src/utils.js [hunk 3]
```diff
@@ -557,6 +558,12 @@ export function isFixedItems(schema) {
   );
 }
 
+export function isCustomWidget(uiSchema) {
+  return (
+    uiSchema.hasOwnProperty("ui:widget") && uiSchema["ui:widget"] !== "hidden"
```
[the remark this answers is not in the document]
Comment (alfonsoar) [reply packages/core/src/utils.js]: done

File: packages/core/src/components/fields/ArrayField.js [hunk 4]
```diff
@@ -532,6 +536,53 @@ class ArrayField extends Component {
...
+      onFocus,
+      formData: items,
+      registry = getDefaultRegistry(),
+      rawErrors,
+      name,
+    } = this.props;
+    const { widgets, formContext } = registry;
+    const title = schema.title || name;
+
+    const { widget = "custom", ...options } = {
```
[the remark this answers is not in the document]
Comment (alfonsoar) [reply packages/core/src/components/fields/ArrayField.js]: done.

File: packages/core/src/utils.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (alfonsoar) [reply packages/core/src/utils.js]: done.

File: packages/core/src/components/fields/ArrayField.js [hunk 5]
```diff
@@ -451,6 +452,9 @@ class ArrayField extends Component {
     if (isMultiSelect(schema, rootSchema)) {
       return this.renderMultiSelect();
     }
+    if (isCustomWidget(uiSchema)) {
+      return this.renderCustomWidget();
+    }
```
[the remark this answers is not in the document]
Comment (alfonsoar) [reply packages/core/src/components/fields/ArrayField.js:452]: arranged, and added an additional test for fixed arrays. files array there was already a test [here](https://github.com/rjsf-team/react-jsonschema-form/blob/1bbd0ada6c886d97030833859e3e92a8ef1a6505/packages/core/test/ArrayField_test.js#L245).

File: CHANGELOG.md (hunk 2 above)
Comment (alfonsoar) [reply CHANGELOG.md]: done.

File: packages/core/src/utils.js [hunk 6]
```diff
@@ -557,6 +558,14 @@ export function isFixedItems(schema) {
   );
 }
 
+export function isCustomWidget(uiSchema) {
+  return (
+    // TODO: Remove the `&& uiSchema["ui:widget"] !== "hidden"` once we support hidden widgets for arrays.
+    // https://react-jsonschema-form.readthedocs.io/en/latest/usage/widgets/#hidden-widgets
+    "widget" in getUiOptions(uiSchema) && uiSchema["ui:widget"] !== "hidden"
```
Comment (epicfaace) [inline packages/core/src/utils.js:565]:
```suggestion
    "widget" in getUiOptions(uiSchema) && getUiOptions(uiSchema)["widget"] !== "hidden"
```

**Review by epicfaace — APPROVED** [review]

File: packages/core/src/utils.js (hunk 6 above)
Comment (epicfaace) [reply packages/core/src/utils.js]: @alfonsoar I made this update

**Pull request merged**